addons/l10n_pe_website_sale/static/src/js/website_sale.js

```diff
@@ -0,0 +1,95 @@
+/** @odoo-module **/
+import {WebsiteSale} from "@website_sale/js/website_sale";
+
+WebsiteSale.include({
+    events: Object.assign({}, WebsiteSale.prototype.events, {
+        "change select[name='state_id']": "_onChangeState",
+        "change select[name='city_id']": "_onChangeCity",
+    }),
+    start: function () {
+        this.selectCities = document.querySelector("select[name='city_id']");
+        this.selectDistricts = document.querySelector("select[name='l10n_pe_district']");
+        this.cityBlock = document.querySelector(".div_city");
+        this.autoFormat = document.querySelector(".checkout_autoformat");
+        this.selectState = document.querySelector("select[name='state_id']");
+        this.isPeruvianCompany = parseInt(document.querySelector("html").getAttribute("data-is-peruvian-company"));
+        if(this.isPeruvianCompany && this.selectState) {
+            let selectCountry = document.querySelector("select[name='country_id']").value;
+            if (this.selectState.value == "" && selectCountry !== '') {
+                this.selectState.options[1].selected = true;
+            }
+        }
+        return this._super.apply(this, arguments);
+    },
+    _changeOption: function (selectCheck, rpcRoute, place, selectElement) {
+        if (!selectCheck) {
+            return;
+        }
+        return this._rpc({
+            route: rpcRoute,
+        }).then((data) => {
+            if (this.isPeruvianCompany) {
+                if (data[place]?.length) {
+                    selectElement.innerHTML = "";
+                    data[place].forEach((item) => {
+                        let opt = document.createElement("option");
+                        opt.textContent = item[1];
+                        opt.value = item[0];
+                        opt.setAttribute("data-code", item[2]);
+                        selectElement.appendChild(opt);
+                    });
+                    selectElement.parentElement.style.display = "block";
+                } else {
+                    selectElement.value = "";
+                    selectElement.parentElement.style.display = "none";
+                }
+            }
+        });
+    },
+    _onChangeState: function () {
+        if (this.isPeruvianCompany) {
+            const rpcRoute = `/shop/state_infos/${this.selectState.value}`;
+            const cities = "cities";
+            const selectCheck = this.selectState.value;
+            return this.autoFormat.length
+                ? this._changeOption(selectCheck, rpcRoute, cities, this.selectCities).then(() => this._onChangeCity())
+                : undefined;
```

Comment:
Handle the delivery address issue here
```suggestion
            if (this.selectState.value === "" && this.selectCountry.value !== '') {
                this.selectState.options[1].selected = true;
            }
            const state = this.selectState.value;
            const rpcRoute = `/shop/state_infos/${state}`;
            return this.autoFormat.length
                ? this._changeOption(state, rpcRoute, "cities", this.selectCities).then(() => this._onChangeCity())
                : undefined;
```